README.md

```diff
@@ -103,28 +103,47 @@ Which can be categorized into three sub sets:
 
 #### Libraries
 
-These Melon modules are:
-- Exchange Adapters
-- Rewards
+They interact with the Melon protocol using as pre-linked libraries to the Melon version contract. These Melon modules are:
+
+- **Exchange Adapters**:
+These are adapters betweens Melon protocol and Decentralized exchanges. Responsible for relaying information, making and taking orders on exchanges.
 
-They interact with the Melon protocol using as pre-linked libraries to the Melon version contract.
+- **Rewards**:
+This module defines functions for calculating Management and Performance rewards for the fund manager. Management reward is calculated on the time managed irrespective of performance of the fund while performance reward is calculated based on the performance.
 
 #### Boolean functions
 
-These Melon modules are:
-- Participation
-- Risk Managment
+They interact with the Melon protocol using boolean functions. That is functions which take a certain set of inputs and return either true or false. These Melon modules are:
+
+- **Participation**:
+It comprises of two primary boolean functions isSubscriptionPermitted and isRedemptionPermitted which enforce rules for investing and redemption from the fund. They take the parameter inputs as specified in the earlier section.
 
-They interact with the Melon protocol using boolean functions. That is functions which take a certain set of inputs and return either true or false.
+- **Risk Managment**:
+It currently comprises of two boolean functions isMakePermitted and isTakePermitted. They take the parameter inputs as specified in the earlier section. This module can be extended with custom logic to prevent malevolent actions by the fund manager. This may include checks if the order price is significantly different from the reference price, e.t.c.
 
 #### Infrastructure
 
-These Melon modules are:
-- Asset registrars
-- Data feeds
+These are security critical infrastructure modules. These Melon modules are:
+
+- **Asset registrars**:
+It is a chain independant asset registrar module. Only the registered assets will be available. Cross-chain asset integration is enabled through Polkadot in future. An asset can be registered via the register function by specifying the following parameters:
 
-These are modules security critical infrastructure modules.
-The reason they are security criticial is that the correctness of the data they provide cannot directly be enforced or guaranteed.
+Name | Data Type | Description
+--- | --- | ---
+asset | `address` | Address of the asset
+name | `string` | Human-readable name of the Asset as in ERC223 token standard
+symbol | `string` | Human-readable symbol of the Asset as in ERC223 token standard
+decimal | `uint256` |  Decimal, order of magnitude of precission, of the Asset as in ERC223 token standard
```

Comment:
for consistency this should be `uint`

if `uint256` used in Asset registrar then it should be changed there as well